### PR TITLE
Implement c# bindings

### LINKED ISSUE #102: Consider adding Interoptopus for additional bindings
Please consider adding support for https://github.com/ralfbiedert/interoptopus/ to support generating bindings for C#

### LINKED ISSUE #146: Update uniffi-bindgen to v0.23.0
There has been a big structural rework on UNiFFI crate with v.0.23.0 and it has breaking changes.

Here's the migration guide:
[Migrating to UniFFI 0.23](https://github.com/mozilla/uniffi-rs/blob/main/CHANGELOG.md#migrating-to-uniffi-023)

### PR DESCRIPTION
**Pull request by roeierez**
Using this uniff-rs extension we are able now to add c# binding: https://github.com/NordSecurity/uniffi-bindgen-cs
Added console application demonstration on how to use and two make targets for mac and linux.
Also upgraded uniffi to 23.0.

fixes #102 
fixes #146

### CODE REVIEW

File: libs/sdk-bindings/makefile [hunk 1]
```diff
@@ -30,19 +30,30 @@ darwin-universal: $(SOURCES)
 	lipo -create -output ../target/darwin-universal/release/libbreez_sdk_bindings.dylib ../target/aarch64-apple-darwin/release/libbreez_sdk_bindings.dylib ../target/x86_64-apple-darwin/release/libbreez_sdk_bindings.dylib
 	lipo -create -output ../target/darwin-universal/release/libbreez_sdk_bindings.a ../target/aarch64-apple-darwin/release/libbreez_sdk_bindings.a ../target/x86_64-apple-darwin/release/libbreez_sdk_bindings.a
 
+csharp-darwin: darwin-universal
```
Comment (erdemyerebasmaz) [inline libs/sdk-bindings/makefile:33]: Should we add `csharp-darwin` & `csharp-linux` targets to `make all`?

**Review by erdemyerebasmaz — APPROVED** [review]
UniFFI migration looks good and I was able to generate bindings for `csharp-darwin` with no issues.

**Review by JssDWt — COMMENTED** [review]
In general this looks pretty good. Although it would be great to make the C# code async if/where possible. Maybe we can ask the developer of the library whether that's an option.

I was ablt to generate the code without any issues.

**Review by ok300 — CHANGES_REQUESTED** [review]
Code LGTM, but IMO the PR is missing #179 

---

Ideally also (but not sure if possible):
* a command, or list of commands, on how to run the provided `test_breez_sdk.cs` locally
* alternatively, a step in the CI that runs this test, or a similar (simpler, if necessary) test

File: libs/sdk-bindings/makefile (hunk 1 above)
Comment (ok300) [reply libs/sdk-bindings/makefile:33]: Sounds like it should belong there.

File: libs/sdk-bindings/makefile [hunk 2]
```diff
@@ -30,19 +30,30 @@ darwin-universal: $(SOURCES)
...
+	cargo install uniffi-bindgen-cs --git https://github.com/NordSecurity/uniffi-bindgen-cs
+	uniffi-bindgen-cs src/breez_sdk.udl -o ffi/csharp -c ./uniffi.toml 
+	cp ../target/darwin-universal/release/libbreez_sdk_bindings.dylib ffi/csharp
+
+TARGET ?= aarch64-unknown-linux-gnu
+csharp-linux: $(SOURCES)
```
Comment (ok300) [inline libs/sdk-bindings/makefile:39]: I tested `make csharp-linux` and it worked.

I did have to change the default target to `x86_64-unknown-linux-gnu`, but each tester can override it with `--target` based on his own local architecture and available rust targets.

File: libs/sdk-bindings/tests/bindings/test_breez_sdk.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
+using breez.breez_sdk;
```
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.cs:2]: Thought it is going to be easy but it is taken by the generator from the udl file which is common to all platforms and the majority of the platforms use lower case as best practice...

**Comment by roeierez** [comment]
> Ideally also (but not sure if possible):
> 
> * a command, or list of commands, on how to run the provided `test_breez_sdk.cs` locally
> * alternatively, a step in the CI that runs this test, or a similar (simpler, if necessary) test

@ok300  Added a test that run the c# code and updated the CI

**Review by ok300 — APPROVED** [review]

File: libs/sdk-bindings/tests/bindings/test_breez_sdk.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (JssDWt) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.cs:2]: I got it to work by changing [this line](https://github.com/NordSecurity/uniffi-bindgen-cs/blob/9de616cc17958f079c9636893dde28867fd73b3c/bindgen/templates/wrapper.cs#L25) to 

```
namespace {{ config.package_name().to_upper_camel_case() }}.{{ ci.namespace().to_upper_camel_case() }};
```

That changes the name to `Breez.BreezSdk`. 
I'm not trying to imply that's what we _have_ to do. But, you know, we are autists right? This snake case gets in the soul of a dotnet developer.

[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.cs:2]: Yes that's the change needed but it is not in our code. I can create a PR in their repo but I don't want to maintain a fork for that 
How about merging and hope that the PR will be approved?

[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.cs:2]: Created a PR:  https://github.com/NordSecurity/uniffi-bindgen-cs/pull/11
Changed to work with this fork and now the namespace is Breez.Sdk as you suggested.

**Pull request merged**